### PR TITLE
Kdbai v1.4

### PR DESCRIPTION
**Pull request by alexgiannak**
Thank you for contributing to LangChain!

- [ ] **PR title**: "package: description"
  - Where "package" is whichever of langchain, community, core, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

- [ ] **PR message**: ***Delete this entire checklist*** and replace with
 This PR updates kdbai vector store to the latest version of the kdbai_client 

- [ ] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [ ] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

**Comment by alexgiannak** [comment]
Hi Langchain Team,

This PR includes the latest updates for the KDBAI integration, bringing it up to compatibility with client-server version 1.4, which is scheduled for release on Monday. Since this version updat introduces breaking changes to the kdbai API, these updates are essential to ensure continued functionality with the new version.

It would be great if you could review and merge this at your earliest convenience.

Thanks so much for your help!

File: libs/community/langchain_community/vectorstores/kdbai.py [hunk 1]
```diff
@@ -34,9 +30,6 @@ def __init__(
         self,
         table: Any,
         embedding: Embeddings,
-        distance_strategy: Optional[
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/kdbai.py:37]: breaking change

File: libs/community/langchain_community/vectorstores/kdbai.py [hunk 2]
```diff
@@ -194,15 +187,16 @@ def similarity_search_with_score(
...
         return self.similarity_search_by_vector_with_score(
-            self._embed_query(query), k=k, filter=filter, **kwargs
+            self._embed_query(query), index=index, k=k, filter=filter, **kwargs
         )
 
     def similarity_search_by_vector_with_score(
         self,
         embedding: List[float],
         *,
+        index: str,
```
Comment (efriis) [inline libs/community/langchain_community/vectorstores/kdbai.py:197]: breaking change

**Review by efriis — COMMENTED** [review]
there's a few breaking changes in here (don't think i marked all of them) that prevent this from being merged.

Also, if you're interested in maintaining this integration without us in the loop, we'd love to get an integration package out! Future PRs against langchain would just be {docs updates, as well as registering your package in `libs/packages.yml`, deprecating this community integration in favor of your integration package}

Here's the guide, and if you have questions, feel free to leave them in the comments on those pages so others can see them! https://python.langchain.com/docs/contributing/how_to/integrations/

**Comment by efriis** [comment]
closing for breaking changes with the recommendation to launch and register an external package for the new verison!

**Pull request closed without merging**